src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -113,4 +115,408 @@ private async Task<HttpResponseMessage> GetDataSet(Guid dataSetId)
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            var dataSetVersionGenerator = DataFixture
+                .DefaultDataSetVersion(
+                    filters: 1,
+                    indicators: 1,
+                    locations: 1,
+                    timePeriods: 3)
+                .WithStatus(dataSetVersionStatus)
+                .WithPublished(DateTimeOffset.UtcNow)
+                .WithDataSetId(dataSet.Id);
+
+            DataSetVersion dataSetVersion = dataSetVersionStatus == DataSetVersionStatus.Unpublished
+                ? dataSetVersionGenerator.WithUnpublished(DateTimeOffset.UtcNow)
+                : dataSetVersionGenerator;
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSetVersions.Add(dataSetVersion));
+
+            var response = await ListVersions(
+                dataSetId: dataSet.Id,
+                page: 1,
+                pageSize: 1);
+
+            var content = response.AssertOk<DataSetVersionPaginatedListViewModel>(useSystemJson: true);
+
+            Assert.NotNull(content);
+            Assert.Equal(1, content.Paging.Page);
+            Assert.Equal(1, content.Paging.PageSize);
+            Assert.Equal(1, content.Paging.TotalResults);
+
+            var result = Assert.Single(content.Results);
+
+            Assert.Equal(dataSetVersion.Version, result.Number);
+            Assert.Equal(dataSetVersion.VersionType(), result.Type);
+            Assert.Equal(dataSetVersion.Status, result.Status);
+            Assert.Equal(
+                dataSetVersion.Published!.Value.ToUnixTimeSeconds(),
+                result.Published.ToUnixTimeSeconds()
+            );
+            Assert.Equal(
+                dataSetVersion.Unpublished?.ToUnixTimeSeconds(),
+                result.Unpublished?.ToUnixTimeSeconds()
+            );
+            Assert.Equal(dataSetVersion.Notes, result.Notes);
+            Assert.Equal(dataSetVersion.TotalResults, result.TotalResults);
+            Assert.Equal(
+                TimePeriodFormatter.Format(
+                    dataSetVersion.MetaSummary.TimePeriodRange.Start.Year,
+                    dataSetVersion.MetaSummary.TimePeriodRange.Start.Code),
+                result.TimePeriods.Start);
+            Assert.Equal(
+                TimePeriodFormatter.Format(
+                    dataSetVersion.MetaSummary.TimePeriodRange.End.Year,
+                    dataSetVersion.MetaSummary.TimePeriodRange.End.Code),
+                result.TimePeriods.End);
+            Assert.Equal(dataSetVersion.MetaSummary.GeographicLevels, result.GeographicLevels);
+            Assert.Equal(dataSetVersion.MetaSummary.Filters, result.Filters);
+            Assert.Equal(dataSetVersion.MetaSummary.Indicators, result.Indicators);
+        }
+
+        [Fact]
+        public async Task DataSetVersionAvailableForOtherDataSet_Returns200_OnlyVersionForRequestedDataSet()
```

Comment:
Similar to other places, could be simpified to `AvailableVersionForOtherDataSet_Returns200_OnlyVersionForRequestedDataSet`